neonvm/controllers/alerttracker/tracker_test.go

```diff
@@ -0,0 +1,80 @@
+package alerttracker_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/neondatabase/autoscaling/neonvm/controllers/alerttracker"
+)
+
+type nowMock struct {
+	ts time.Time
+}
+
+func (n *nowMock) Now() time.Time {
+	return n.ts
+}
+
+func (n *nowMock) Add(d time.Duration) {
+	n.ts = n.ts.Add(d)
+}
+
+func newNowMock() *nowMock {
+	ts, _ := time.Parse("2006-01-02", "2024-01-01")
+	return &nowMock{ts: ts}
+}
+
+func TestTracker(t *testing.T) {
+	now := newNowMock()
+	alerttracker.Now = now.Now
```

Comment:
This seems sketchy... (does it produce data races if we run tests in parallel?)

I would recommend something more like what we went with in `pkg/util/recentcnt[_test].go`

Reply:
That can work too; I would still rather avoid dependency injection if we can, but wouldn't hold up approval over it.